src/ace/managers/sprite.c

```diff
@@ -170,11 +183,50 @@ void spriteSetBitmap(tSprite *pSprite, tBitMap *pBitmap) {
 		return;
 	}
 
-	pSprite->pBitmap = pBitmap;
-	spriteSetHeight(pSprite, pBitmap->Rows - 2);
+	if (pBitmap->Depth == 2) {
+		pSprite->pBitmap = pBitmap;
+		spriteSetHeight(pSprite, pBitmap->Rows - 2);
 
-	tSpriteChannel *pChannel = &s_pChannelsData[pSprite->ubChannelIndex];
-	spriteChannelRequestCopperUpdate(pChannel);
+		tSpriteChannel *pChannel = &s_pChannelsData[pSprite->ubChannelIndex];
+		spriteChannelRequestCopperUpdate(pChannel);
+	}
+	if (pBitmap->Depth == 4) {
+
+		pSprite->is4PP=1;
+
+		tBitMap *pointers_low;
+		tBitMap *pointers_high;
+
+        pointers_high = bitmapCreate(
+            ubByteWidth*8, pBitmap->Rows,
+            2, BMF_CLEAR | BMF_INTERLEAVED);
+
+        pointers_low = bitmapCreate(
+            ubByteWidth*8, pBitmap->Rows,
+            2, BMF_CLEAR | BMF_INTERLEAVED);
```

Comment:
This is a big no no. We can't have any Create functions inside functions which are potentially called in main game loop (e.g. when trying to animate sprite).

The reason for this is that bitmapCreate does allocate memory, and that causes OS to be turned back on. It causes significant performance drop because turning on OS takes 2+ frames.